psignifit/tests/test_matlab.py

```diff
@@ -0,0 +1,77 @@
+"""
+The Matlab version of psignifit-3 was extensively tested with
+thousands of simulations for publishing the original paper
+(Schütt, Harmeling, Macke, & Wichmann; 2016). Thus we consider
+the matlab results as ground-truth and test for regression against them.
+"""
+from pathlib import Path
+import json
+import random
+
+import numpy as np
+import pytest
+
+import psignifit
+from psignifit._matlab import param_pydict2matlist
+
+CASE_DIR = Path(__file__).parent / 'matlab_test_cases'
+
+
+def yield_test_cases():
+    return (file.name.rsplit('_', maxsplit=1)[0] for file in sorted(CASE_DIR.glob('test_*_data.csv')))
+
+
+def yield_test_data(cases=yield_test_cases()):
+    for name in cases:
+        dataset = CASE_DIR / f'{name}_data.csv'
+        options = CASE_DIR / f'{name}_opt.json'
+        results = CASE_DIR / f'{name}_res.json'
+
+        data = np.loadtxt(dataset, delimiter=',')
+        with open(options, 'r') as f:
+            options = json.load(f)
+        with open(results, 'r') as f:
+            results = json.load(f)
+        yield data, options, results
+
+
+def compare_with_matlab(data, options, results):
+    config = psignifit.Configuration.from_matlab_options(options, raise_matlab_only=False)
+    pyresults = psignifit.psignifit(data, conf=config)
+    # Testing for similarity:
+    # allclose(actual, desired) = abs(actual - desired) <= atol + rtol * abs(desired)
+    # ==> atol=0.001 assures equality of small differences
+    #     rtol=1 assures the same order of magnitude
+    #
+    # These matlab results are currently not tested:
+    # 'X1D', 'logPmax', 'integral', 'marginals', 'marginalsX', 'marginalsW', 'data',
+    # 'devianceResiduals', 'deviance', 'Cov', 'Cor', 'timestamp', 'meanFit', 'MAP'
+
+    ## POINT ESTIMATE
+    np.testing.assert_allclose(param_pydict2matlist(pyresults.parameter_fit),
+                               results['Fit'],
+                               atol=0.001, rtol=1)
+
+    # INTERVAL ESTIMATE
+    mat_ci = results['conf_Intervals']
+    if options['expType'] == 'equalAsymptote':
+        # matlab returns gamma filled with Nones, but python expects gamma==lambda
+        mat_ci[3] = mat_ci[2]
+    # matlab dims are (n_param, 2, n_ci), but python expects (n_param, n_ci, 2),
+    mat_ci = np.transpose(mat_ci, (0, 2, 1))
+    py_ci = param_pydict2matlist(pyresults.confidence_intervals)
+    np.testing.assert_allclose(py_ci, mat_ci, atol=0.001, rtol=2)
+
+
+@pytest.mark.skip(reason="Testing all cases takes approx 30min, should be run explicitly.")
```

Comment:
Marking this tests as `skip` means that they are not run, not even in the continuous integration workflow. Should we add a pytest marker like `slow`, or a customer one like `matlab`, so that we can run these tests selectively from the command line?